gui/elements/qt/_layer_buttons.py

```diff
@@ -57,35 +53,24 @@ def dropEvent(self, event):
         event.accept()
 
 class QAddLayerButton(QPushButton):
-    def __init__(self, layers):
+    def __init__(self):
         super().__init__()
 
-        self.layers = layers
         self.setIcon(QIcon(path_add))
         self.setFixedWidth(28)
         self.setFixedHeight(28)
         self.setToolTip('Add layer')
-        self.clicked.connect(self.on_click)
         styleSheet = """QPushButton {background-color:lightGray; border-radius: 3px;}
             QPushButton:pressed {background-color:rgb(0, 153, 255); border-radius: 3px;}
             QPushButton:hover {background-color:rgb(0, 153, 255); border-radius: 3px;}"""
         self.setStyleSheet(styleSheet)
 
-    def on_click(self):
-        if self.layers.viewer.dimensions.max_dims == 0:
-            empty_markers = empty((0, 2))
-        else:
-            empty_markers = empty((0, self.layers.viewer.dimensions.max_dims))
-        self.layers.viewer.add_markers(empty_markers)
-
 class QAnnotationCheckBox(QCheckBox):
-    def __init__(self, layers):
+    def __init__(self):
```

Comment:
Unless I am thinking about this all wrong (which I'll admit is quite probable!), this is the exact opposite of what @royerloic was advocating: the gui elements don't know about the model, but the model knows all about the GUI elements.

Reply:
yup it seems like my change takes us in the wrong direction, have posted some more comments / questions in #92 to resolve before I fix

Reply:
I have reverted this